### PR TITLE
feat: add service api documentation

### PR DESCRIPTION
**Pull request by sbckr**

### CODE REVIEW

**Comment by sbckr** [comment]
This update makes use of https://github.com/blueswen/mkdocs-swagger-ui-tag to visualize OpenAPI Documents.

File: config/nav.yml [hunk 1]
```diff
@@ -6,21 +6,26 @@
...
+              - Manual:
+                  - getting-started/deployment/manual/index.md
+                  - Platform Setup: getting-started/deployment/manual/platform-setup.md
+                  - Stack: getting-started/deployment/manual/stack.md
+              - Infrastructure as Code:
+                  - getting-started/deployment/infrastructure-as-code/index.md
+                  - Local: getting-started/deployment/infrastructure-as-code/local.md
+          - Configuring the CLI: getting-started/cli.md
+          - Millionaires Problem: getting-started/millionaires.md
+      - Service API:
```
Comment (strieflin) [inline config/nav.yml:25]: I think, we should add an overview page here that describes:

1. What is described in the subpages.
2. That Open API is used
3. That this is rather interesting for people that want to implement a client. Others should rather use the existing clients.

Comment (strieflin) [inline config/nav.yml:25]: I would suggest the following page hiearchy: Reference -> API Specification -> Caster / Amhora / Ephmeral

File: docs/openapi/oad/amphora.oad.yaml [hunk 2]
```diff
@@ -0,0 +1,856 @@
+openapi: 3.1.0
+info:
+  title: Amphora Service API
+  description: |
+    This API allows to interact with the Amphora service. It provides functionality to store, fetch and manage secret
+    shared data as used for MPC-based distributed computation on encrypted data in the Carbyne Stack platform.
+
+    The endpoints provided by Amphora are divided into three categories:
+    - User facing: \
+      Used by clients for secret sharing, accessing, and managing secret shares and their tags (*/input-masks*,
```
Comment (strieflin) [inline docs/openapi/oad/amphora.oad.yaml:10]: What is the point of enumerating the paths here?

File: docs/openapi/oad/amphora.oad.yaml [hunk 3]
```diff
@@ -0,0 +1,856 @@
+openapi: 3.1.0
+info:
+  title: Amphora Service API
+  description: |
+    This API allows to interact with the Amphora service. It provides functionality to store, fetch and manage secret
+    shared data as used for MPC-based distributed computation on encrypted data in the Carbyne Stack platform.
+
+    The endpoints provided by Amphora are divided into three categories:
+    - User facing: \
```
Comment (strieflin) [inline docs/openapi/oad/amphora.oad.yaml:9]: To be consistent maybe add a "(`/`)" here.

File: docs/openapi/oad/amphora.oad.yaml [hunk 4]
```diff
@@ -0,0 +1,856 @@
...
+      Used by clients for secret sharing, accessing, and managing secret shares and their tags (*/input-masks*,
+      */masked-input*, */secret-shares*).
+    - Intra VCP (`/intra-vcp`): \
+      Used by services running in the same Virtual Cloud Provider to access locally stored secret shares, e.g. for
+      ingesting into a computation.
+    - Inter VCP (`/inter-vcp`): \
```
Comment (strieflin) [inline docs/openapi/oad/amphora.oad.yaml:15]: Can we have a blank line in between the points here?

File: docs/openapi/oad/amphora.oad.yaml [hunk 5]
```diff
@@ -0,0 +1,856 @@
...
+  description: |
+    This API allows to interact with the Amphora service. It provides functionality to store, fetch and manage secret
+    shared data as used for MPC-based distributed computation on encrypted data in the Carbyne Stack platform.
+
+    The endpoints provided by Amphora are divided into three categories:
+    - User facing: \
+      Used by clients for secret sharing, accessing, and managing secret shares and their tags (*/input-masks*,
+      */masked-input*, */secret-shares*).
+    - Intra VCP (`/intra-vcp`): \
+      Used by services running in the same Virtual Cloud Provider to access locally stored secret shares, e.g. for
```
Comment (strieflin) [inline docs/openapi/oad/amphora.oad.yaml:13]: services -> Carbyne Stack microservices

File: docs/openapi/oad/amphora.oad.yaml [hunk 6]
```diff
@@ -0,0 +1,856 @@
...
+      Used for communication with other Virtual Cloud Providers in the same Virtual Cloud.
+  version: 0.1.0
+  license:
+    name: Apache-2.0
+    url: https://github.com/carbynestack/amphora/blob/0b1866f9fbe1619729fc16fd8f9c7af679eaf979/LICENSE
+  contact:
+    name: Carbyne Stack
+    url: https://carbynestack.io/
+tags:
+  - name: User facing
```
Comment (strieflin) [inline docs/openapi/oad/amphora.oad.yaml:25]: I think it should be "User-facing", "Inter-VCP", "Intra-VCP"

File: docs/openapi/oad/amphora.oad.yaml [hunk 7]
```diff
@@ -0,0 +1,856 @@
...
+            format: uuid
+      responses:
+        200:
+          description: OK
+        400:
+          description: Bad Request
+          content:
+            text/plain:
+              schema:
+                type: string
```
Comment (strieflin) [inline docs/openapi/oad/amphora.oad.yaml:57]: Shouldn't there be an explanation of what the string is that is returned here (see also many other places in the spec)?

File: docs/openapi/oad/castor.oad.yaml [hunk 8]
```diff
@@ -0,0 +1,390 @@
...
+          description: Internal Server Error
+          content:
+            text/plain:
+              schema:
+                type: string
+  /intra-vcp/telemetry:
+    get:
+      summary: GET intra-vcp/telemetry
+      description: |
+        Retrieves the telemetry data (consumption per second and available tuples) for all types of tuple managed by
```
Comment (strieflin) [inline docs/openapi/oad/castor.oad.yaml:104]: tuple -> tuples

File: docs/openapi/oad/castor.oad.yaml [hunk 9]
```diff
@@ -0,0 +1,390 @@
+openapi: 3.1.0
+info:
+  title: Castor Service API
+  description: |
+    This API allows to interact with the Castor service. It provides methods to reserve and consume tuples, and to get
+    telemetry data. The service endpoints are separated into two categories: inter-VCP and intra-VCP.
```
Comment (strieflin) [inline docs/openapi/oad/castor.oad.yaml:6]: "inter-VCP and intra-VCP" is not needed here. It's part of the following enumeration already.

File: docs/openapi/oad/castor.oad.yaml [hunk 10]
```diff
@@ -0,0 +1,390 @@
...
+
+    - Inter VCP (`/inter-vcp`): \
+      Endpoints for communication with other Virtual Cloud Providers in the same Virtual Cloud.
+    - Intra VCP (`/intra-vcp`): \
+      Endpoints that are to be consumed by the VCP internal services only. These endpoints are not intended to be
```
Comment (strieflin) [inline docs/openapi/oad/castor.oad.yaml:11]: "intended to be" should be removed

File: docs/openapi/oad/ephemeral.oad.yaml [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    url: https://github.com/carbynestack/ephemeral/blob/7ce9f830de3c5852afdc6c3c6c591fc5d04a50f7/LICENSE
+  contact:
+    name: Carbyne Stack
+    url: https://carbynestack.io/
+paths:
+  /:
+    post:
+      summary: Run a computation
+      description: |
+        Start a disctributed computation. The result of the computation is stored as a new Amphora Secret referenced
```
Comment (strieflin) [inline docs/openapi/oad/ephemeral.oad.yaml:18]: I would rather say: Trigger execution of an MPC function.

File: docs/openapi/oad/ephemeral.oad.yaml [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+        by the gameID provided when triggering the computation.
+      operationId: activate
+      tags:
+        - ephemeral
+      parameters:
+        - name: compile
+          description: |
+            Indicates whether *new* mpc program code is provided with this request. If a program is provided, ot will
```
Comment (strieflin) [inline docs/openapi/oad/ephemeral.oad.yaml:26]: ot -> it

Comment (strieflin) [inline docs/openapi/oad/ephemeral.oad.yaml:26]: mpc -> MPC

File: docs/openapi/oad/ephemeral.oad.yaml
Comment (strieflin) [inline docs/openapi/oad/ephemeral.oad.yaml:1]: Schemas are missing for the body is missing (as given for the Amphora, Castor APIs)

File: docs/openapi/oad/amphora.oad.yaml
Comment (strieflin) [inline docs/openapi/oad/amphora.oad.yaml:1]: The API version "0.1" seems to be wrong. They should match the service release version. This is the case also for the other OADs.

**Review by strieflin — CHANGES_REQUESTED** [review]

File: config/nav.yml (hunk 1 above)
Comment (sbckr) [reply config/nav.yml]: done

Comment (sbckr) [reply config/nav.yml]: new hierarchy applied

File: docs/openapi/oad/amphora.oad.yaml (hunk 2 above)
Comment (sbckr) [reply docs/openapi/oad/amphora.oad.yaml]: removed

File: docs/openapi/oad/amphora.oad.yaml (hunk 3 above)
Comment (sbckr) [reply docs/openapi/oad/amphora.oad.yaml]: Added

File: docs/openapi/oad/amphora.oad.yaml (hunk 4 above)
Comment (sbckr) [reply docs/openapi/oad/amphora.oad.yaml]: Added a line between the individual points for the Amphora OAD as well as Castor OAD

File: docs/openapi/oad/amphora.oad.yaml (hunk 5 above)
Comment (sbckr) [reply docs/openapi/oad/amphora.oad.yaml]: changed

File: docs/openapi/oad/amphora.oad.yaml (hunk 6 above)
Comment (sbckr) [reply docs/openapi/oad/amphora.oad.yaml]: Changed

File: docs/openapi/oad/amphora.oad.yaml (hunk 7 above)
Comment (sbckr) [reply docs/openapi/oad/amphora.oad.yaml]: introduced schema `ErrorMessage`

File: docs/openapi/oad/castor.oad.yaml (hunk 8 above)
Comment (sbckr) [reply docs/openapi/oad/castor.oad.yaml]: fixed

File: docs/openapi/oad/castor.oad.yaml (hunk 10 above)
Comment (sbckr) [reply docs/openapi/oad/castor.oad.yaml]: Done

File: docs/openapi/oad/ephemeral.oad.yaml (hunk 11 above)
Comment (sbckr) [reply docs/openapi/oad/ephemeral.oad.yaml]: Done. Also replaced `program` with `function` in general.

File: docs/openapi/oad/ephemeral.oad.yaml (hunk 12 above)
Comment (sbckr) [reply docs/openapi/oad/ephemeral.oad.yaml]: Done

Comment (sbckr) [reply docs/openapi/oad/ephemeral.oad.yaml]: Done for all occurrences (for ephemeral, castor and amphora)

File: docs/openapi/oad/amphora.oad.yaml
Comment (sbckr) [reply docs/openapi/oad/amphora.oad.yaml:1]: Updated versions to latest release

File: docs/openapi/oad/ephemeral.oad.yaml
Comment (sbckr) [reply docs/openapi/oad/ephemeral.oad.yaml:1]: Explicitly introduced the Schema

**Review by strieflin — CHANGES_REQUESTED** [review]

File: docs/reference/api-specification/index.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+
+The REST API documentation serves as a valuable resource for individuals
+interested in implementing a client application or developing a new microservice
+that leverages the capabilities of Carbyne Stack. By studying the API
+description, developers can gain insights into the available functionality,
+understand the expected input and output formats, and effectively integrate
+their applications with Carbyne Stack.
+
+However, it is important to note that for most users, leveraging the existing
+clients provided is the recommended approach. These clients have already
```
[the remark this answers is not in the document]
Comment (sbckr) [reply docs/reference/api-specification/index.md:24]: rephrased

File: docs/reference/api-specification/index.md [hunk 14]
```diff
@@ -0,0 +1,29 @@
+# API Specification
+
+The Carbyne Stack platform provides a comprehensive set of microservices,
+working together to create a fully functional MPC-party. To facilitate
+interaction and communication, each  service exposes a REST API. This API serves
+as a standardized interface for accessing and utilizing the functionalities
+provided by the individual services, to facilitate both, inter and intra VCP
```
[the remark this answers is not in the document]
Comment (sbckr) [reply docs/reference/api-specification/index.md:7]: fixed

**Review by strieflin — APPROVED** [review]
LGTM

**Pull request merged**